TKGame/Animations/ScreenTransistion.cs

```diff
@@ -0,0 +1,108 @@
+using StbImageSharp;
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Data;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TKGame.Animations
+{
+    public class ScreenTransition
+    {
+        #region Member Data
+        private Rectangle[] sourceRectangles;
+
+        private static readonly int screenWidth = 1600;
+        private static readonly int screenHeight = 900;
+        private int spriteX = 509;
+        private int spriteY = 285;
+        private int buffer = 5;
+        byte currentIndex = 0;
+        private float timer = 0;
+        private int threshold = 50;
+        private bool loop = true;
+
+
+
+        #endregion
+        /// <summary>
+        /// creates ScreenTransition Object
+        /// </summary>
+        /// <param name="graphicsDevice"></param>
+        public ScreenTransition (GraphicsDevice graphicsDevice)
+        {
+            //individual sprite images
+            sourceRectangles = new Rectangle[16];
+            sourceRectangles[1] = new Rectangle(buffer, buffer, spriteX, spriteY);
+            sourceRectangles[2] = new Rectangle(spriteX +(buffer*2), buffer, spriteX, spriteY);
+            sourceRectangles[3] = new Rectangle(buffer, spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[4] = new Rectangle(spriteX + (buffer*2), spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[5] = new Rectangle((spriteX  * 2) + (buffer * 2), spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[6] = new Rectangle(buffer, (spriteY *2) + (buffer), spriteX, spriteY);
+            sourceRectangles[7] = new Rectangle((spriteX + buffer) + 2, (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[8] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[9] = new Rectangle(buffer, (spriteY * 3) + (2 * buffer) , spriteX, spriteY);
+            sourceRectangles[10] = new Rectangle((spriteX + buffer) + 2, (spriteY * 3) + (2 * buffer), spriteX, spriteY);
+            sourceRectangles[11] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY * 3) + (6 * buffer), spriteX, spriteY);
+            sourceRectangles[12] = new Rectangle(buffer, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[13] = new Rectangle((spriteX + buffer) + 2, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[14] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[15] = new Rectangle(buffer, (spriteY + buffer) * 5, spriteX, spriteY);
+            sourceRectangles[0] = new Rectangle((spriteX + buffer) + 2, (spriteY +buffer) * 5 + 16, spriteX, spriteY);
```

Comment:
if I'm not mistaken you could do something like:
```csharp
sourceRectangles = new Rectangle[16]
{
    { x1, y1, w1, h1 },
    { x2, y2, w2, h2 },
    ....
};
```
i can't remember exactly with arrays specifically, but i have to imagine C# can hook you up. i mean for now it works lol